js/client-chat.js

```diff
@@ -185,10 +185,14 @@
 		// highlight
 
 		getHighlight: function (message) {
-			var highlights = Tools.prefs('highlights') || [];
-			if (!app.highlightRegExp) {
+			var highlights = Tools.prefs('highlights') || {};
+			if (highlights.constructor === 'Array') this.convertHighlights();
+			var thisroom = (this.type === 'battle' ? 'battlehls' : this.id);
+			if (!app.highlightRegExp || !app.highlightRegExp[(Config.server.id + 'globalhls')] || !app.highlightRegExp[(Config.server.id + thisroom)]) {
 				try {
-					this.updateHighlightRegExp(highlights);
+					if (!app.highlightRegExp) app.highlightRegExp = [];
+					if (!app.highlightRegExp['globalhls']) this.updateHighlightRegExp(highlights[(Config.server.id + 'globalhls')], 'globalhls');
```

Comment:
Why are you calling them `globalhls` and `battlehls`? It seems better to call them `global` and `battle`, since those are reserved names, while `globalhls` and `battlehls` could be actual room IDs.